app/bundles/LeadBundle/EventListener/ReportSubscriber.php

```diff
@@ -162,86 +142,21 @@ public function onReportBuilder(ReportBuilderEvent $event)
-                ],
-                'l.points' => [
-                    'label' => 'mautic.lead.points',
-                    'type'  => 'int',
-                ],
-                'l.owner_id' => [
-                    'label' => 'mautic.lead.report.owner_id',
-                    'type'  => 'int',
-                    'link'  => 'mautic_user_action',
-                ],
-                'u.first_name' => [
-                    'label' => 'mautic.lead.report.owner_firstname',
-                    'type'  => 'string',
-                ],
-                'u.last_name' => [
-                    'label' => 'mautic.lead.report.owner_lastname',
-                    'type'  => 'string',
-                ],
-            ];
-
-            $leadFields = $this->fieldModel->getEntities([
-                'filter' => [
-                    'force' => [
-                        [
-                            'column' => 'f.object',
-                            'expr'   => 'like',
-                            'value'  => 'lead',
-                        ],
-                    ],
-                ],
-            ]);
-
+            $columns        = $this->fieldsBuilder->getLeadFieldsColumns('l.');
             $companyColumns = $this->companyReportData->getCompanyData();
 
-            $filters = $columns = array_merge(
-                $columns,
-                $companyColumns,
-                $this->getFieldColumns($leadFields, 'l.')
-            );
-
-            // Append segment filters
-            $userSegments = $this->listModel->getUserLists();
-            $list         = [];
-            foreach ($userSegments as $segment) {
-                $list[$segment['id']] = $segment['name'];
-            }
-            $filters['s.leadlist_id'] = [
-                'alias'     => 'segment_id',
-                'label'     => 'mautic.core.filter.lists',
-                'type'      => 'select',
-                'list'      => $list,
-                'operators' => [
-                    'eq' => 'mautic.core.operator.equals',
-                ],
-            ];
-
-            $filters['l.owner_id'] = [
-                'label' => 'mautic.lead.list.filter.owner',
-                'type'  => 'select',
-                'list'  => $this->userModel->getRepository()->getUserList('', 0),
-            ];
+            $filters = $this->fieldsBuilder->getLeadFilter('l.', 's.');
 
             $data = [
                 'display_name' => 'mautic.lead.leads',
-                'columns'      => $columns,
-                'filters'      => $filters,
+                'columns'      => array_merge(
+                    $columns,
+                    $companyColumns
+                ),
+                'filters' => array_merge(
+                    $filters,
+                    $companyColumns
+                ),
             ];
 
             $event->addTable(self::CONTEXT_LEADS, $data, self::GROUP_CONTACTS);
```

Comment:
I didn't catch this in the company report PR. We are hard coding the context to CONTEXT_LEADS here but we need to call addTable for each of the contexts within `$this->leadContexts`. Otherwise, the columns are not injected into each of the data sources. 

Same in other places where multiple contexts are applicable. 

For example in lines 180 
```
if ($event->checkContext([self::CONTEXT_LEADS, self::CONTEXT_LEAD_POINT_LOG])) {
    // Add shared graphs
    $event->addGraph(self::CONTEXT_LEADS, 'line', 'mautic.lead.graph.line.leads');
```
addGraph needs to be called for both `self::CONTEXT_LEADS` and `self::CONTEXT_LEAD_POINT_LOG`

Then line 189 only needs to check for the frequency rule context
```
if ($event->checkContext(self::CONTEXT_CONTACT_FREQUENCYRULES)) {
```

Right now $companyContexts contains a single context so it's fine, but if we add more, 205-209 would need to be set for each context.